driver/annotations.md

```diff
@@ -0,0 +1,171 @@
+objects. It serves to establish the role of some internal keys in the native AST
+into specific key roles in the UAST. This is needed so the rule applying process
+knows what parts of the native AST use to do the translation. The skeleton of this
+struct will be automatically generated by the `bblfsh-sdk init` command. 
+
+The type and meaning of the fields in the native.ObjectToNoder struct can be
+checked on the [GoDoc documentation of the
+type](https://godoc.org/github.com/bblfsh/sdk/protocol/native#ObjectToNoder).
+
+### UASTParserBuilder function
+
+This automatically generated function will return a driver.UASTParser object that
+will be constructed using your provided ToNoder implementation. This is the
+function referred in the main.go file and you usually will not need to change it.
+
+### Examples
+
+Check the already implemented [Java
+ObjectToNoder](https://github.com/bblfsh/java-driver/blob/master/driver/normalizer/parser.go)
+or [Python
+one](https://github.com/bblfsh/python-driver/blob/master/driver/normalizer/parser.go)
+for real world examples of the ObjectToNoder struct filled.
+
+## File: annotation.go
+
+### AnnotationRules 
+
+This variable will hold a Rule object and is the result of the applying of the
+different rules using the functions defined in the file ``sdk/uast/ann/ann.go``
+file. This is done by calling the `On` rule constructor and chaining calls to the
+different selectors, rule constructors (a rule can hold other rules itself) and
+predicates to build your AST to UAST transformator.
+
+To see how rules are built it's important to understand that a rule is composed
+of:
+
+- One or more *predicates* that will select what nodes will match the rule.
+- One of more *actions* that will apply to the matching nodes.
+- Zero or more *selectors* that will optionally select more nodes starting
+  from the current ones and apply the rule/s in its argument list.
+- Zero or more additional *rules* contained inside to the *selectors*.
+
+The rule constructor is the function `On(predicates...) *Rule`. As you can see
+in the signature, it takes one of more predicates and returns a rule pointer. This 
+allows for chaining of different methods taking a rule pointer. A general form
+for a rule definition could this be like:
+
+```go
+// Simple example:
+var r  := On(somePredicate).SomeAction(actionParams)
+
+// A more elaborate example:
+var r2 := On(otherPredicate).SomeSelector(
+    On(thirdPredicate).SomeAction(actionParams), // embedded rule
+    On(fouthPredicate).SomeSelector( // embedded rule with its own childs
+        On(fithPredicate).SomeAction(actionParams),
+    )).SomeAction(actionPArams)
+```
+
+#### Predicates
+
+Predicates are any function that follow the signature: 
+
+```go
+func(n *uast.Node) bool
+```
+
+Where uast.Node are the Nodes that will be iterated during the annotation
+process. You could define your own predicate functions but the ann.go file
```

Comment:
If the normal usage doesn't need the user to build its own predicate, it shouldn't be the first thing they read. I'd rephrase:

`ann.go` fail has a list of functions with the predicates you'll probably need:
- `Any()`
- ...

These predicate functions should be enough for most cases, but you can still define your own if needed.